scripts/zkai_d128_snark_receipt_timing_setup_gate.py

```
@@ -0,0 +1,646 @@
+        artifact.get("source_public_payload_sha256"),
+        sha256_bytes(canonical_json_bytes(load_json(SOURCE_PUBLIC))),
+        "source public payload hash",
+        layer="artifact_binding",
+    )
+    expect_equal(artifact.get("source_proof_bytes"), SOURCE_PROOF.stat().st_size, "source proof size", layer="artifact_binding")
+    expect_equal(artifact.get("source_verification_key_bytes"), SOURCE_VK.stat().st_size, "source vk size", layer="artifact_binding")
+    generated_public_hashes = require_list(artifact.get("generated_public_file_sha256_values"), "generated public hashes", layer="artifact_binding")
+    expect_equal(generated_public_hashes, [sha256_file(SOURCE_PUBLIC)], "generated public hashes", layer="artifact_binding")
+    proof_size_values = require_list(artifact.get("generated_proof_size_bytes_values"), "proof size values", layer="artifact_binding")
+    if proof_size_values != metrics.get("proof_size_bytes_values") or any(not isinstance(value, int) or value <= 0 for value in proof_size_values):
+        raise D128SnarkTimingSetupError("proof size values mismatch", layer="artifact_binding")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)